src/lib/main/dialog_view.js

```diff
@@ -150,6 +169,158 @@ function updateAutoscanDescription(sliderValue) {
     AutoscanSliderDescriptions[sliderValue];
 }
 
+const ScanModeMap = new Map([
+  ['anywhere', {
+    description: '',
+    options: {
+      partialScan: false,
+      contentMode: Page.contentModeEnum.TEXT,
+      matchMode: Page.matchModeEnum.FIRST,
+    },
+  }],
+  ['inside-elements', {
+    description: `Check only inside selected elements using HTML elements 
+    selector.`,
+    options: {
+      partialScan: true,
+      requireExactMatchCount: true,
+      contentMode: Page.contentModeEnum.TEXT,
+      matchMode: Page.matchModeEnum.FIRST,
+    },
+  }],
+  ['count-only', {
+    description: `Check only for change in number of HTML element matches.
+    Content is ignored.`,
+    options: {
+      partialScan: true,
+      requireExactMatchCount: true,
+      contentMode: Page.contentModeEnum.IGNORE,
+      matchMode: Page.matchModeEnum.FIRST,
+    },
+  }],
+]);
```

Comment:
There are no invalid combinations of attributes! Any combination outside of the allowed values is still valid (unless it contains wrong type or something like that).

Reply:
I spent quite a bit of time designing and thinking about this and I must say I am very strongly against the way you propose. The way I designed it is that the options are actually only front end thing for the user. It might add a bit of perceived complexity to the page attributes and an extra resolve function to the UI but it makes for a better code for scanning functions. Also it's easier to extend in the future.

If you had scan mode like you propose you would either need to have duplicate code or complex conditions. Neither of which is particularly appealing.

Reply:
Ok, that's fair enough. Thanks for the explanation.